gtsfm/two_view_estimator.py

```diff
@@ -215,6 +215,68 @@ def bundle_adjust(
 
         return i2Ti1_optimized.rotation(), Unit3(i2Ti1_optimized.translation()), valid_corr_idxs
 
+    def __get_2view_report_from_results(
+        self,
+        i2Ri1_computed: Optional[Rot3],
+        i2Ui1_computed: Optional[Unit3],
+        keypoints_i1: Keypoints,
+        keypoints_i2: Keypoints,
+        verified_corr_idxs: np.ndarray,
+        inlier_ratio_wrt_estimate: float,
+        gt_camera_i1: Optional[gtsfm_types.CAMERA_TYPE],
+        gt_camera_i2: Optional[gtsfm_types.CAMERA_TYPE],
+        gt_scene_mesh: Optional[Any],
+    ) -> TwoViewEstimationReport:
+        """Generate a TwoViewEstimationReport from the results of the two-view estimation.
+
+        Currently metrics wrt GT camera only supports cases where gt_camera is PinholeCameraCal3Bundler.
+
+        Args:
+            i2Ri1_computed: computed relative rotation.
+            i2Ui1_computed: computed relative unit translation.
+            keypoints_i1: keypoints from image i1.
+            keypoints_i2: keypoints from image i2.
+            verified_corr_idxs: indices of verified correspondences between i1 and i2.
+            inlier_ratio_wrt_estimate: inlier ratio w.r.t. the estimated relative pose.
+            gt_camera_i1: ground truth camera for i1.
+            gt_camera_i2: ground truth camera for i2.
+            gt_scene_mesh: ground truth scene mesh.
+
+        Returns:
+            TwoViewEstimationReport object, some fields may be None if either gt_camera are None.
+        """
+        if gt_camera_i1 and gt_camera_i2:
+            # if we have the expected GT data, evaluate the computed relative pose
+            R_error_deg, U_error_deg = compute_relative_pose_metrics(
+                i2Ri1_computed, i2Ui1_computed, gt_camera_i1.pose(), gt_camera_i2.pose()
+            )
+            # TODO: add support for other camera models
+            if isinstance(gt_camera_i1, PinholeCameraCal3Bundler) and isinstance(
+                gt_camera_i2, PinholeCameraCal3Bundler
+            ):
+                inlier_mask_wrt_gt, reproj_error_wrt_gt = metric_utils.compute_correspondence_metrics(
+                    keypoints_i1,
```

Comment:
nit: can we use keyword args here since we're changing the keyword order and bc we got so many args rn here?